changelogs/unreleased/enforce-mypy.yml

```diff
@@ -0,0 +1,3 @@
+description: Enforece the use of mypy using mypy baseline
+change-type: patch
+destination-branches: [master]
```

Comment:
I think we will need this on iso7 as well, to prevent merge hell. 
(but might be separate MR, due to different baseline )